src/thor/route_action.cc

```diff
@@ -395,38 +427,33 @@ void thor_worker_t::path_arrive_by(Api& api, const std::string& costing) {
   GraphId first_edge;
   std::unordered_map<size_t, std::pair<EdgeTrimmingInfo, EdgeTrimmingInfo>> vias;
   std::vector<thor::PathInfo> path;
-  auto& correlated = *api.mutable_options()->mutable_locations();
   std::vector<std::string> algorithms;
-  api.mutable_trip()->mutable_routes()->Reserve(api.options().alternates() + 1);
+  const Options& options = api.options();
+  valhalla::Trip& trip = *api.mutable_trip();
+  trip.mutable_routes()->Reserve(options.alternates() + 1);
 
-  // For each pair of locations
-  for (auto origin = ++correlated.rbegin(); origin != correlated.rend(); ++origin) {
+  auto route_two_points = [&, this](auto& origin, auto& destination) -> bool {
     // Get the algorithm type for this location pair
-    auto destination = std::prev(origin);
-    thor::PathAlgorithm* path_algorithm =
-        get_path_algorithm(costing, *origin, *destination, api.options());
+    thor::PathAlgorithm* path_algorithm = get_path_algorithm(costing, *origin, *destination, options);
     path_algorithm->Clear();
     algorithms.push_back(path_algorithm->name());
     LOG_INFO(std::string("algorithm::") + path_algorithm->name());
 
     // TODO: delete this and send all cases to the function above
```

Comment:
so since we deleted this, then can we delete that function i had lying around above and also delete this comment?